pkg/util/provider/machinecontroller/machine_test.go

```diff
@@ -346,6 +362,34 @@ var _ = Describe("machine", func() {
 					err:        false,
 				},
 			}),
+			Entry("machineClass without Finalizer", &data{
```

Comment:
are we just trying to show that machineClass which doesn't have finalizer will be marked by finalizer in this Testcase.

Reply:
The idea is - if a MachineClass does not have a finaliser then the machineClass should not be reconciled.